### PR TITLE
fix(AllEntitiesTable): show Tags

### LINKED ISSUE #31009: Tags column in a specific tag's page is not populated
### Bug description

Repro steps: 
Make sure the feature flag for TAGGING_SYSTEM is enabled and your user has permission to create tags 
1. Visit the tags page
2. Create 2 or more tags 
3. Tag a chart or dashboard with at least 2 tags 
4. From the Tags page, click on an individual tag to see all assets tagged with that tag 

Note that the tags column should show all OTHER tags (not including the tag whose page you are viewing) but it is blank 

### Screenshots/recordings

image

### Superset version

master / latest-dev

### Python version

3.9

### Node version

16

### Browser

Chrome

### Additional context

_No response_

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

### PR DESCRIPTION
**Pull request by alexandrusoare**
Fixes #31009

### SUMMARY
Fixed bug where Tags column in a specific tag's page is not populated.
The problem was a typo.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before 
before

After
after

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by sadpandajoe** [comment]
@alexandrusoare thanks for this PR, do you mind adding tests for this to make sure in the future when someone makes edits to this it won't break again?

File: superset-frontend/src/features/allEntities/AllEntitiesTable.tsx [hunk 1]
```diff
@@ -119,7 +121,8 @@ export default function AllEntitiesTable({
               <TagsList
                 tags={tags.filter(
                   (tag: Tag) =>
-                    tag.type === 'TagTypes.custom' || tag.type === 1,
+                    (tag.type === 'TagType.custom' && tag.id !== tagId) ||
+                    (tag.type === 1 && tag.id !== tagId),
```
Comment (geido) [inline superset-frontend/src/features/allEntities/AllEntitiesTable.tsx:125]: Can we simplify this logic? Something like this:

```
<TagsList
  tags={tags.filter((tag: Tag) => 
    ['TagType.custom', 1].includes(tag.type) && tag.id !== tagId
  )}
  maxTags={MAX_TAGS_TO_SHOW}
/>
```

**Review by geido — COMMENTED** [review]
Thanks! Could you please update the description of the PR to mention what was the cause of the error?

**Review by geido — APPROVED** [review]
Thanks for the fix and the test!

**Pull request merged**